src/Workspaces/Core/Portable/LinkedFileDiffMerging/LinkedFileDiffMergingLogger.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Internal.Log;
+using static Microsoft.CodeAnalysis.LinkedFileDiffMergingSession;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal class LinkedFileDiffMergingLogger
+    {
+        private const string Id = nameof(Id);
+        private static ConcurrentDictionary<int, LogAggregator> LogAggregators = new ConcurrentDictionary<int, LogAggregator>();
```

Comment:
also, if you do want to use this, I think it would be better to set concurrency level to concurrentDictionary otherwise, it is known to create a lot of objects to reduce as much as contention possible.

also, you need to do something about holding onto workspace strongly. it can leak.